NullpointerException when running a query and the dataset is not found

#### Environment details

1. BigQuery
2. OS type and version: Mate Desktop Environment 1.20.1
3. Java version: Java 8
4. bigquery version(s): 1.16.0

#### Steps to reproduce

  1. Running a query when the dataset is not found

#### Code example

```java
      QueryJobConfiguration dmlQueryConfig = QueryJobConfiguration.newBuilder(dmlQuery).build();

      // Execute the query.
      TableResult result = bigquery.query(dmlQueryConfig);
```

#### Stack trace
![image](https://user-images.githubusercontent.com/11322055/92620935-9b70cb00-f299-11ea-8013-7c14390a81b1.png)

#### Any additional information below
![image](https://user-images.githubusercontent.com/11322055/92620522-200f1980-f299-11ea-91e4-822507da9bdc.png)
![image](https://user-images.githubusercontent.com/11322055/92620664-47fe7d00-f299-11ea-94c4-e4be3596c8b4.png)

Following these steps guarantees the quickest resolution possible.

Thanks!

@emanuelbatista you can see below example and errors which i am getting when dataset is not exists. if i am doing wrong in below example, could you please provide an example or correct below example so i can reproduce that error ?

```JAVA
public class QueryExample {
  public static void main(String[] args) throws InterruptedException {
    String dmlQuery = "SELECT * FROM test_dataset_123.insert_rows";
    BigQuery bigQuery = BigQueryOptions.getDefaultInstance().getService();
    QueryJobConfiguration dmlQueryConfig = QueryJobConfiguration.newBuilder(dmlQuery).build();

    // Execute the query.
    TableResult result = bigQuery.query(dmlQueryConfig);
    result.iterateAll().forEach(val -> System.out.println(val));
  }
}
```
```JAVA
Exception in thread "main" com.google.cloud.bigquery.BigQueryException: Not found: Dataset project-xyz:test_dataset_123 was not found in location US
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:106)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.getQueryResults(HttpBigQueryRpc.java:586)
	at com.google.cloud.bigquery.BigQueryImpl$34.call(BigQueryImpl.java:1208)
	at com.google.cloud.bigquery.BigQueryImpl$34.call(BigQueryImpl.java:1203)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.bigquery.BigQueryImpl.getQueryResults(BigQueryImpl.java:1202)
	at com.google.cloud.bigquery.BigQueryImpl.getQueryResults(BigQueryImpl.java:1186)
	at com.google.cloud.bigquery.Job$1.call(Job.java:329)
	at com.google.cloud.bigquery.Job$1.call(Job.java:326)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.poll(RetryHelper.java:64)
	at com.google.cloud.bigquery.Job.waitForQueryResults(Job.java:325)
	at com.google.cloud.bigquery.Job.getQueryResults(Job.java:291)
	at com.google.cloud.bigquery.BigQueryImpl.query(BigQueryImpl.java:1173)
	at com.qlogic.QueryExample.main(QueryExample.java:15)
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 404 Not Found
GET https://www.googleapis.com/bigquery/v2/projects/project-xyz/queries/dcaa3caf-e55f-49ad-8e44-c5099fd5a3bc?location=US&maxResults=0
{
  "code" : 404,
  "errors" : [ {
    "domain" : "global",
    "message" : "Not found: Dataset project-xyz:test_dataset_123 was not found in location US",
    "reason" : "notFound"
  } ],
  "message" : "Not found: Dataset project-xyz:test_dataset_123 was not found in location US",
  "status" : "NOT_FOUND"
}
	at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:150)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:113)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:40)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:444)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1108)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:542)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:475)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:592)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.getQueryResults(HttpBigQueryRpc.java:584)
	... 16 more

```

--- a/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java
+++ b/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java
@@ -297,8 +297,10 @@ public class Job extends JobInfo {
       job = reload();
     }
     if (job.getStatus() != null && job.getStatus().getError() != null) {
-      throw new JobException(
-          getJobId(), ImmutableList.copyOf(job.getStatus().getExecutionErrors()));
+      throw new BigQueryException(
+          job.getStatus().getExecutionErrors() == null
+              ? ImmutableList.of(job.getStatus().getError())
+              : ImmutableList.copyOf(job.getStatus().getExecutionErrors()));
     }
 
     // If there are no rows in the result, this may have been a DDL query.

